ui/app/components/input-number.js

```diff
@@ -1,56 +1,52 @@
 const Component = require('react').Component
 const h = require('react-hyperscript')
 const inherits = require('util').inherits
+const { addCurrencies } = require('../conversion-util')
 
 module.exports = InputNumber
 
 inherits(InputNumber, Component)
 function InputNumber () {
   Component.call(this)
 
-  this.state = {
-    value: 0,
-  }
-
   this.setValue = this.setValue.bind(this)
 }
 
 InputNumber.prototype.componentWillMount = function () {
-  const { initValue = 0 } = this.props
+  const { value = 0 } = this.props
 
-  this.setState({ value: initValue })
+  this.setState({ value: Number(value) })
 }
 
 InputNumber.prototype.setValue = function (newValue) {
-  const { fixed, min = -1, onChange } = this.props
+  const { fixed, min = -1, max = Infinity, onChange } = this.props
 
-  if (fixed) newValue = Number(newValue.toFixed(4))
+  newValue = Number(fixed ? newValue.toFixed(4) : newValue)
 
-  if (newValue >= min) {
-    this.setState({ value: newValue })
+  if (newValue >= min && newValue <= max) {
     onChange(newValue)
   }
 }
 
 InputNumber.prototype.render = function () {
-  const { unitLabel, step = 1, placeholder } = this.props
-  const { value } = this.state
+  const { unitLabel, step = 1, placeholder, value = 0 } = this.props
+  const valueAsNum = Number(value)
```

Comment:
See above: not sure why we need this conversion / if we need it, why it needs to be a separate line. Can just be in L39.    I like reusing state in L18 though.

Reply:
Removed in 32098a64